### ISSUE TITLE
didn't set vue-eslint-parser, but checked the rules defined in eslint-plugin-vue

### ISSUE THREAD

**Issue by irenetang1993** [vuejs/eslint-plugin-vue#2378]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**   7.32.0
- **eslint-plugin-vue version:**  8.7.1
- **Vue version:**  2.6.14
- **Node version:**  18.16.1
- **Operating System:** macOS Ventura 13.4.1

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    'plugin:vue/essential',
    'plugin:prettier/recommended',
  ],
  parserOptions: {
    parser: '@babel/eslint-parser',
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
  },
};

```

**What did you do?**	

```vue
<template>
  <MyComponent :foo="abc" foo="dd" />
</template>
```

**What did you expect to happen?**
https://eslint.vuejs.org/user-guide/#what-is-the-use-the-latest-vue-eslint-parser-error
From this link I learned that when using eslint-plugin-vue, we should set vue-eslint-parser as eslint parser. If I don't set vue-eslint-parser, it doesn't check the rules defined in eslint-plugin-vue.
To test, I used eslint-plugin-vue but didn't set vue-eslint-parser, which I thought wouldn't report the error, but it still prompted the vue/no-duplicate-attributes error.

**What actually happened?**

don't prompted the vue/no-duplicate-attributes error.
**Repository to reproduce this issue**

**Comment by FloEdelmann** [comment]
This use case is not supported in this plugin. Why would you want to use eslint-plugin-vue without vue-eslint-parser?
